cmd/end2endtest/helpers.go

```diff
@@ -631,3 +631,53 @@ func cspGenProof(pid, voterKey []byte, csp *ethereum.SignKeys) (*apiclient.Censu
 
 	return &apiclient.CensusProof{Proof: caProofBytes}, nil
 }
+
+func (t *e2eElection) verifyVoteCount(startTime time.Time, nvotesExpected int) error {
+	// Wait for all the votes to be verified
+	log.Infof("waiting for all the votes to be registered...")
+	for {
+		count, err := t.api.ElectionVoteCount(t.election.ElectionID)
+		if err != nil {
+			log.Warn(err)
+		}
+		if count == uint32(nvotesExpected) {
+			break
+		}
+		time.Sleep(time.Second * 5)
+		log.Infof("verified %d/%d votes", count, nvotesExpected)
+		if time.Since(startTime) > t.config.timeout {
+			return fmt.Errorf("timeout waiting for votes to be registered")
+		}
+	}
+
+	log.Infof("%d votes registered successfully, took %s (%d votes/second)",
+		t.config.nvotes, time.Since(startTime), int(float64(nvotesExpected)/time.Since(startTime).Seconds()))
+
+	return nil
+}
+
+func (t *e2eElection) setElectionEnd() error {
+	// Set the account back to the organization account
+	if err := t.api.SetAccount(t.config.accountPrivKeys[0]); err != nil {
+		return fmt.Errorf("cannot set the account back to the organization %w", err)
+	}
+
+	// End the election by setting the status to ENDED
+	log.Infof("ending election...")
+	if _, err := t.api.SetElectionStatus(t.election.ElectionID, "ENDED"); err != nil {
+		return fmt.Errorf("cannot set election status to ENDED %w", err)
+	}
+	return nil
+}
+
+func (t *e2eElection) waitUntilElectionResults() (*vapi.ElectionResults, error) {
+	// Wait for the election to be in RESULTS state
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*100)
```

Comment:
if you take the suggestion of refactoring `c.timeout` flag, you can replace this arbitrary `time.Second*100` with `c.timeout`

Reply:
Using `c.timeout` (which would default to apiclient.WaitTimeout = 3 blocks = 24s) seems not enough time for election publish final results, so I update to use `t.config.timeout*3`